ts/WoltLabSuite/WebComponent/woltlab-core-pagination.ts

```diff
@@ -0,0 +1,316 @@
+      if (nextLinkElement) {
+        nav.append(nextLinkElement);
+      }
+    }
+
+    #getNavElement(): HTMLElement {
+      const nav = document.createElement("nav");
+      nav.setAttribute("role", "navigation");
+      nav.setAttribute("aria-label", window.WoltLabLanguage.getPhrase("wcf.page.pagination"));
+      nav.classList.add(this.#className);
+
+      return nav;
+    }
+
+    #getPreviousLinkElement(): HTMLDivElement | undefined {
+      if (this.page === 1) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(`${this.#className}__prev`);
+
+      const button = this.#getButtonElement(this.page - 1);
+      if (button instanceof HTMLAnchorElement) {
+        button.rel = "prev";
+      }
+      button.title = window.WoltLabLanguage.getPhrase("wcf.global.page.previous");
+      button.classList.add("jsTooltip");
+      div.append(button);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-left");
+      button.append(icon);
+
+      return div;
+    }
+
+    #getNextLinkElement(): HTMLDivElement | undefined {
+      if (this.page === this.count) {
+        return;
+      }
+
+      const div = document.createElement("div");
+      div.classList.add(`${this.#className}__next`);
+
+      const button = this.#getButtonElement(this.page + 1);
+      if (button instanceof HTMLAnchorElement) {
+        button.rel = "next";
+      }
+      button.title = window.WoltLabLanguage.getPhrase("wcf.global.page.next");
+      button.classList.add("jsTooltip");
+      div.append(button);
+
+      const icon = document.createElement("fa-icon");
+      icon.setIcon("arrow-right");
+      button.append(icon);
+
+      return div;
+    }
+
+    #getButtonElement(page: number): HTMLAnchorElement | HTMLButtonElement {
+      let button: HTMLAnchorElement | HTMLButtonElement;
+      const url = this.getLinkUrl(page);
+
+      if (url) {
+        button = document.createElement("a");
+        button.href = url;
+      } else {
+        button = document.createElement("button");
+        button.type = "button";
+
+        if (this.page === page) {
+          button.disabled = true;
+        } else {
+          button.addEventListener("click", () => {
+            this.#switchPage(page);
+          });
+        }
+      }
+
+      button.classList.add(`${this.#className}__link`);
+
+      return button;
+    }
+
+    #getLinkItem(page: number): HTMLLIElement {
+      const li = document.createElement("li");
+      li.classList.add(`${this.#className}__item`);
+
+      const button = this.#getButtonElement(page);
+      button.setAttribute("aria-label", window.WoltLabLanguage.getPhrase("wcf.page.pageNo", { pageNo: page }));
+      if (page === this.page) {
+        button.setAttribute("aria-current", "page");
+        button.classList.add(`${this.#className}__link--current`);
+      }
+      button.textContent = page.toString();
```

Comment:
```suggestion
      button.textContent = page.toLocaleString(document.documentElement.lang);
```
For correct formatting of numbers > 1000.